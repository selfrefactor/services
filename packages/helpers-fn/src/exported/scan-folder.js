const { existsSync } = require('fs')
const { fdir } = require('fdir')
const defaultFilterFn = x => x.endsWith('.js')
const defaultExcludeFn = x => x.includes('node_modules') || x.startsWith('.')

async function scanFolder({
  folder,
  filterFn = defaultFilterFn,
  excludeFn = defaultExcludeFn,
  maxDepth = 4,
}){
  if (!existsSync(folder)){
    throw new Error(`${ folder } - folder path is wrong as it doesn't exist`)
  }

  const files = await new fdir()
    .withMaxDepth(maxDepth)
    .withFullPaths()
    .exclude(excludeFn)
    .filter(filterFn)
    .crawl(folder)
    .withPromise()

  return files
}

exports.scanFolder = scanFolder
