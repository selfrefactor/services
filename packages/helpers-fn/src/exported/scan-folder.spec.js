import { resolve } from 'path'

import { scanFolder } from './scan-folder'
const testDir = resolve(__dirname, '../../src')

test('happy', async () => {
  expect(await scanFolder({ folder : testDir })).toMatchSnapshot()
})

test('with filter', async () => {
  const filterFn = x => x.endsWith('.spec.js')

  expect(await scanFolder({
    folder : testDir,
    filterFn,
  })).toMatchSnapshot()
})

test('with exclude', async () => {
  const excludeFn = dir => dir.endsWith('log')

  expect(await scanFolder({
    folder : testDir,
    excludeFn,
  })).toMatchSnapshot()
})
