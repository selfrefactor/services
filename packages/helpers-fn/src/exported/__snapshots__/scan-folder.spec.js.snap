// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy 1`] = `
[
  "/home/w/repos/services/packages/helpers-fn/src/index.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/copy.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/copy.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/create-benchmark.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/create-complex-benchmark.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/default-to.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/default-to.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/exec.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/get-staged-files.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/get-staged-files.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/run-tests.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/run-tests.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/scan-folder.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/scan-folder.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/log/constants.js",
  "/home/w/repos/services/packages/helpers-fn/src/log/log.js",
  "/home/w/repos/services/packages/helpers-fn/src/log/log.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/tests/benchmark-utils.js",
  "/home/w/repos/services/packages/helpers-fn/src/tests/create-benchmark.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/utils/parse-monitor-data.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/utils/parse-monitor-data.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/log/_modules/big-log.js",
  "/home/w/repos/services/packages/helpers-fn/src/log/_modules/box.js",
  "/home/w/repos/services/packages/helpers-fn/src/log/_modules/colorized-background.js",
  "/home/w/repos/services/packages/helpers-fn/src/log/_modules/colorized-text.js",
  "/home/w/repos/services/packages/helpers-fn/src/log/_modules/log-object.js",
  "/home/w/repos/services/packages/helpers-fn/src/log/_modules/log-with-icon.js",
  "/home/w/repos/services/packages/helpers-fn/src/log/_modules/separator.js",
  "/home/w/repos/services/packages/helpers-fn/src/tests/benchmarks/allPass.js",
]
`;

exports[`with exclude 1`] = `
[
  "/home/w/repos/services/packages/helpers-fn/src/index.js",
  "/home/w/repos/services/packages/helpers-fn/src/tests/benchmark-utils.js",
  "/home/w/repos/services/packages/helpers-fn/src/tests/create-benchmark.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/copy.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/copy.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/create-benchmark.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/create-complex-benchmark.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/default-to.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/default-to.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/exec.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/get-staged-files.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/get-staged-files.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/run-tests.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/run-tests.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/scan-folder.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/scan-folder.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/tests/benchmarks/allPass.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/utils/parse-monitor-data.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/utils/parse-monitor-data.spec.js",
]
`;

exports[`with filter 1`] = `
[
  "/home/w/repos/services/packages/helpers-fn/src/log/log.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/copy.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/default-to.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/get-staged-files.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/run-tests.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/scan-folder.spec.js",
  "/home/w/repos/services/packages/helpers-fn/src/exported/utils/parse-monitor-data.spec.js",
]
`;
